src/components/Drawer/Drawer.types.ts

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+import { DrawerSizes } from './Drawer.enums';
+
+export type Sizes = typeof DrawerSizes[keyof typeof DrawerSizes];
+
+export interface DrawerProps {
+  children: React.ReactNode;
+  onClose: React.MouseEventHandler;
+  footer?: React.ReactNode;
+  title: string;
+  size?: Sizes;
+  adornment?: React.ReactNode;
+  hasBackdrop?: boolean;
+  dataTestId?: string;
```

Comment:
We should drop this from type, this will be handled by `[key: string]: unknown;`